examples/usb_serial.rs

```diff
@@ -0,0 +1,94 @@
+//! CDC-ACM serial port example using polling in a busy loop.
+//! Target board: Blue Pill
+#![no_std]
+#![no_main]
+
+extern crate panic_semihosting;
+
+use cortex_m::asm::delay;
+use cortex_m_rt::entry;
+use embedded_hal::digital::v2::OutputPin;
+use stm32f1xx_hal::usb::{Peripheral, UsbBus};
+use stm32f1xx_hal::{prelude::*, stm32};
+use usb_device::prelude::*;
+use usbd_serial::{SerialPort, USB_CLASS_CDC};
+
+#[entry]
+fn main() -> ! {
+    let dp = stm32::Peripherals::take().unwrap();
+
+    let mut flash = dp.FLASH.constrain();
+    let mut rcc = dp.RCC.constrain();
+
+    let clocks = rcc
+        .cfgr
+        .use_hse(8.mhz())
+        .sysclk(48.mhz())
+        .pclk1(24.mhz())
+        .freeze(&mut flash.acr);
+
+    assert!(clocks.usbclk_valid());
+
+    // Configure the on-board LED (PC13, green)
+    let mut gpioc = dp.GPIOC.split(&mut rcc.apb2);
+    let mut led = gpioc.pc13.into_push_pull_output(&mut gpioc.crh);
+    led.set_high(); // Turn off
+
+    let mut gpioa = dp.GPIOA.split(&mut rcc.apb2);
+
+    // BluePill board has a pull-up resistor on the D+ line.
+    // Pull the D+ pin down to send a RESET condition to the USB bus.
+    let mut usb_dp = gpioa.pa12.into_push_pull_output(&mut gpioa.crh);
+    usb_dp.set_low();
+    delay(clocks.sysclk().0 / 100);
```

Comment:
Do we need that? I've removed it from my keyboard, and it still work fine.

Reply:
This code is needed only for development. Without it host doesn't reset your device when you upload new firmware.

Reply:
Maybe that could be documented in the comment above.